Support compilation with significantly stronger compiler warning flags.

benalexau opened the pull request:
A project I'm using with ``statsd-cpp`` uses significantly stronger warning flags. It is usually recommended to use strong warning flags where possible (eg Item 1, "Compile cleanly at high warning levels" of C++ Coding Standards by Herb Sutter and Andrei Alexandrescu).

This pull request adds the warning flags I use for C++11 projects and makes the minimal changes required to resolve the resulting warnings from GCC 5.1.0.

euskadi31 commented:
Hello Ben,

Thanks for your Pull Request !

The Travis test failed (see https://travis-ci.org/euskadi31/statsd-cpp/builds/65174879) please fix this problem.

Axel

benalexau commented:
Hi Axel

I removed the extra warning flags from ``CMakeLists.txt`` so it wouldn't fail with other earlier GCC versions or other compilers. I added the additional flags to the commit message though. I also squashed the commit and forced the push so the PR would remain a single commit.

Thanks
Ben

euskadi31 commented:
Hi Ben,

Thank you very much for your pull request !

Axel
